controllers/networkchaos/netem/types.go

```diff
@@ -310,3 +327,175 @@ func mergeNetem(spec v1alpha1.NetworkChaosSpec) (*pb.Netem, error) {
 	}
 	return merged, nil
 }
+
+func (r *Reconciler) applyNetem(ctx context.Context, sources, targets []v1.Pod, networkchaos *v1alpha1.NetworkChaos) error {
+
+	g := errgroup.Group{}
+
+	for index := range sources {
+		pod := &sources[index]
+
+		key, err := cache.MetaNamespaceKeyFunc(pod)
+		if err != nil {
+			return err
+		}
+
+		networkchaos.Finalizers = utils.InsertFinalizer(networkchaos.Finalizers, key)
+	}
+
+	// if we don't specify targets, then sources pods apply netem on all egress traffic
+	if len(targets) == 0 {
+		r.Log.Info("apply netem", "sources", sources)
+		for index := range sources {
+			pod := &sources[index]
+
+			g.Go(func() error {
+				parent := &pb.TcHandle{
+					Major: 1,
+					Minor: 0,
+				}
+
+				handle := &pb.TcHandle{
+					Major: 1,
+					Minor: 0,
+				}
+				return r.applyPod(ctx, pod, networkchaos, parent, handle)
+			})
+		}
+		return g.Wait()
+	}
+
+	// create ipset contains all target ips
+	dstIpset := ipset.BuildIpSet(targets, networkchaos, ipsetPostFix)
+	r.Log.Info("apply netem with filter", "sources", sources, "ipset", dstIpset)
+
+	for index := range sources {
+		pod := &sources[index]
+
+		g.Go(func() error {
+			// NOTE: this is core logic of filtering:
+			// we need handle both filtered traffic with netem actions like delay, loss etc,
+			// and keep rest traffic as normal.
+			// To filter traffic, use tc filter to classify traffic and route to netem qdisc.
+			// Because the default qdisc pfifo_fast is classless, which we can't use here,
+			// using prio with proper priomap as a replacement of pfifo_fast.
```

Comment:
Since we want apply netem on part of the traffic, the traffic divide into two kinds, one with netem qdisc and the other kind is normal traffic.

The normal traffic, linux default is using `prio_fast`, which contains 3 bands, like below:
![image](https://user-images.githubusercontent.com/2183720/79677152-a8c0e680-8220-11ea-9c6a-fe92d39c3085.png)

To maintain the normal traffic behavior in our case, I use `prio` to replace `prio_fast`, those two are very similar but `prio` can have sub qdisc. The traffic will looks like this:
![image](https://user-images.githubusercontent.com/2183720/79677126-6ac3c280-8220-11ea-9d66-2c56a8e8340a.png)
classid 1:[1-3] here is identical to `prio_fast` classid 1:[1-3].
and the priomap here `"1", "2", "2", "2", "1", "2", "0", "0", "1", "1", "1", "1", "1", "1", "1", "1"` is identical to `prio_fast` priomap.